python/ray/services.py

```diff
@@ -577,13 +579,17 @@ def start_reaper():
     Returns:
         ProcessInfo for the process that was started.
     """
+    if ray.util.set_psigdeath(None):
+        # No need for a reaper; already fate-sharing in kernel-mode
+        return None
     # Make ourselves a process group leader so that the reaper can clean
     # up other ray processes without killing the process group of the
     # process that started us.
     try:
         os.setpgrp()
-    except OSError as e:
-        if e.errno == errno.EPERM and os.getpgrp() == os.getpid():
+    except (AttributeError, OSError) as e:
```

Comment:
What is the `AttributeError` check for here?

Reply:
It's because `os.setpgrp` will fail on Windows as the function doesn't exist.

Reply:
Ah I see. In that case shouldn't we just not call it on windows?

Reply:
It's the unguarded fallback case for if for whatever reason the kernel-level fate-sharing doesn't work, so I figured I'd make it work regardless.